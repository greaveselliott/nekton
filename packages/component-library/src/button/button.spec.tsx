import * as React from 'react';
import { shallow } from 'enzyme';
import { Button } from '.';

describe('Button', () => {
  it('renders as expected', () => {
    expect(
      shallow(<Button type="round" style="secondary" size="small" />)
    ).toMatchSnapshot();
  });

  it('renders a link as expected', () => {
    expect(
      shallow(<Button href="#" type="default" style="secondary" />)
    ).toMatchSnapshot();
  });

  it('renders a button elemement as default', () => {
    expect(shallow(<Button />)).toContainMatchingElement('button');
  });

  it('renders an anchor tag if given a href', () => {
    expect(shallow(<Button href="#" />)).toContainMatchingElement('a');
  });

  it('is able to call a function on click', () => {
    const mockOnClickFunction = jest.fn();
    const button = shallow(<Button onClick={mockOnClickFunction} />);
    button.simulate('click');
    expect(mockOnClickFunction).toHaveBeenCalled();
  });

  it('can be disabled', () => {
    expect(shallow(<Button disabled />)).toBeDisabled();
  });

  it('can render children', () => {
    expect(shallow(<Button disabled>abc</Button>)).toIncludeText('abc');
  });
});
