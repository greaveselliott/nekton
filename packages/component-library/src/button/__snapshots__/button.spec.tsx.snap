// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a link as expected 1`] = `
<a
  className="button button--secondary button--default"
  href="#"
/>
`;

exports[`Button renders as expected 1`] = `
<button
  className="button button--secondary button--round button--round--small"
/>
`;
